packages/overlays/src/OverlayController.js

```diff
@@ -735,21 +735,19 @@ export class OverlayController extends EventTargetShim {
       case 'before-show':
         this.__bodyClientWidth = document.body.clientWidth;
         this.__bodyClientHeight = document.body.clientHeight;
-        this.__bodyMarginRight = 0;
-        this.__bodyMarginBottom = 0;
+        this.__bodyMarginRightInline = document.body.style.marginRight;
+        this.__bodyMarginBottomInline = document.body.style.marginBottom;
         break;
       case 'show': {
-        if (supportsCSSTypedObject) {
-          // @ts-expect-error types attributeStyleMap not available yet
-          this.__bodyMarginRight = document.body.computedStyleMap().get('margin-right').value;
-          // @ts-expect-error types computedStyleMap not available yet
-          this.__bodyMarginBottom = document.body.computedStyleMap().get('margin-bottom').value;
-        } else if (window.getComputedStyle) {
+        if (window.getComputedStyle) {
           const bodyStyle = window.getComputedStyle(document.body);
           if (bodyStyle) {
             this.__bodyMarginRight = parseInt(bodyStyle.getPropertyValue('margin-right'), 10);
             this.__bodyMarginBottom = parseInt(bodyStyle.getPropertyValue('margin-bottom'), 10);
           }
+        } else {
+          this.__bodyMarginRight = 0;
```

Comment:
Now that this is wrapped in an else here, this.__bodyMarginRight can be undefined, but we're using it in a computation a bit further down on line 756 and 757, so we get a type error now. Do you think it still makes sense to set these properties to 0 by default even if getComputedStyle is not on window?

Reply:
bodyStyle is actually always defined (I don't see a case where it wouldn't be if `window.getComputedStyle` exists). So the `if (bodyStyle)` on line 744 is useless, which the compiler can't know, and hence throws an error because it believes this.__bodyMarginRight can be undefined.
I'll fix this right now by removing this `if (bodyStyle)` on line 744.

Reply:
Hm yeah fair enough, I think getComputedStyle always returns a CSSStyleDeclaration. If not, a typecast could work in this case, I think it would be accceptable for document.body:

```js
const bodyStyle = /** @type {CSSStyleDeclaration} */ (window.getComputedStyle(document.body));
```